src/frontend/components/UI/OfflineMessage/index.tsx

```diff
@@ -0,0 +1,53 @@
+import ContextProvider from 'frontend/state/ContextProvider'
+import React, { MouseEvent, useContext, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+
+import './index.css'
+
+const OfflineMessage = () => {
+  const { t } = useTranslation()
+  const { connectivity } = useContext(ContextProvider)
+  const [showHint, setShowHint] = useState(false)
+
+  // render nothing if online
+  if (connectivity.status === 'online') {
+    return <></>
+  }
+
+  let content = t('offline-message.offline', 'Offline')
+
+  if (connectivity.status === 'check-online') {
+    if (connectivity.retryIn) {
+      content += t('offline-message.retry-in', {
+        defaultValue: 'Retrying in ... {{seconds}} seconds',
+        seconds: connectivity.retryIn
+      })
+    } else {
+      content = t('offline-message.retrying', 'Retrying')
+    }
+  }
+
+  const toggleHint = (ev: MouseEvent) => {
+    ev.preventDefault()
+    setShowHint(!showHint)
+  }
+
+  const hintHtml = t('offline-message.hint', {
+    defaultValue:
+      'We are checking the connectivity against:{{newline}}github.com,{{newline}}gog.com and{{newline}}store.epicgames.com',
+    newline: '<br />'
+  })
+
+  return (
+    <div className="offline-message">
+      <span>{content}</span>
+      <button onClick={toggleHint}>?</button>
```

Comment:
It will be easier for the user if they just have to hover their mouse to see the modal. This was working for me:

```
<button onMouseEnter={toggleHint} onMouseLeave={toggleHint}>
```

Reply:
I changed this to work with css only on hover instead of this or the click event, less js to maintain